src/backend/cdb/cdbpartition.c

```diff
@@ -7130,6 +7143,17 @@ atpxPartAddList(Relation rel,
 				((CreateStmt *) q)->ownerid = ownerid;
 			}
 
+			StdRdOptions ao_opts;
+
+			if (gp_add_partition_inherits_table_setting && RelationIsAppendOptimized(rel))
+			{
+				ao_opts = *currentAOStorageOptions();
+				resetDefaultAOStorageOpts();
+			}
```

Comment:
Can we don't change new partition options in ProcessUtility internals instead of temporary changes of the ao_storage_opts variable?

Reply:
done